src/pages/settings/Security/AddDelegate/UpdateDelegateRole/UpdateDelegateRolePage.tsx

```diff
@@ -0,0 +1,88 @@
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {clearDelegateRolePendingAction, requestValidationCode, updateDelegateRoleOptimistically} from '@libs/actions/Delegate';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {DelegateRole} from '@src/types/onyx/Account';
+
+type UpdateDelegateRolePageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.SETTINGS.DELEGATE.UPDATE_DELEGATE_ROLE>;
+
+function UpdateDelegateRolePage({route}: UpdateDelegateRolePageProps) {
+    const {translate} = useLocalize();
+    const login = route.params.login;
+    const [currentRole, setCurrentRole] = useState(route.params.currentRole);
+
+    const styles = useThemeStyles();
+    const roleOptions = Object.values(CONST.DELEGATE_ROLE).map((role) => ({
+        value: role,
+        text: translate('delegate.role', {role}),
+        keyForList: role,
+        alternateText: translate('delegate.roleDescription', {role}),
+        isSelected: role === currentRole,
+    }));
+
+    useEffect(() => {
+        updateDelegateRoleOptimistically(login ?? '', currentRole as DelegateRole);
+        return () => clearDelegateRolePendingAction(login);
+        // eslint-disable-next-line react-compiler/react-compiler
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [login]);
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            testID={UpdateDelegateRolePage.displayName}
+        >
+            <HeaderWithBackButton
+                title={translate('delegate.accessLevel')}
+                onBackButtonPress={() => Navigation.goBack()}
+            />
+            <SelectionList
+                isAlternateTextMultilineSupported
+                alternateTextNumberOfLines={4}
+                initiallyFocusedOptionKey={currentRole}
+                shouldUpdateFocusedIndex
+                headerContent={
+                    <Text style={[styles.ph5, styles.pb5, styles.pt3]}>
+                        <>
+                            {translate('delegate.accessLevelDescription')}{' '}
+                            <TextLink
+                                style={[styles.link]}
+                                href={CONST.COPILOT_HELP_URL}
+                            >
+                                {translate('common.learnMore')}
+                            </TextLink>
+                        </>
+                    </Text>
+                }
+                onSelectRow={(option) => {
+                    if (option.isSelected) {
+                        Navigation.dismissModal();
+                        return;
+                    }
+
+                    requestValidationCode();
+                    setCurrentRole(option.value);
```

Comment:
When the option is selected user is required to authenticate using magic code, which means this option is not selected yet. Thus when user just goes back without authenticating this option will be selected even though user did not complete required steps. https://github.com/Expensify/App/pull/53677/files